packages/authenticator/amplify_authenticator/lib/src/utils/dial_code.dart

```diff
@@ -0,0 +1,509 @@
+  static const es = DialCode._('ES', '34', DialCodeResolverKey.es$);
+  static const lk = DialCode._('LK', '94', DialCodeResolverKey.lk$);
+  static const sd = DialCode._('SD', '249', DialCodeResolverKey.sd$);
+  static const sr = DialCode._('SR', '597', DialCodeResolverKey.sr$);
+  static const sj = DialCode._('SJ', '47', DialCodeResolverKey.sj$);
+  static const se = DialCode._('SE', '46', DialCodeResolverKey.se$);
+  static const ch = DialCode._('CH', '41', DialCodeResolverKey.ch$);
+  static const sy = DialCode._('SY', '963', DialCodeResolverKey.sy$);
+  static const tw = DialCode._('TW', '886', DialCodeResolverKey.tw$);
+  static const tj = DialCode._('TJ', '992', DialCodeResolverKey.tj$);
+  static const tz = DialCode._('TZ', '255', DialCodeResolverKey.tz$);
+  static const th = DialCode._('TH', '66', DialCodeResolverKey.th$);
+  static const tl = DialCode._('TL', '670', DialCodeResolverKey.tl$);
+  static const tg = DialCode._('TG', '228', DialCodeResolverKey.tg$);
+  static const tk = DialCode._('TK', '690', DialCodeResolverKey.tk$);
+  static const to = DialCode._('TO', '676', DialCodeResolverKey.to$);
+  static const tt = DialCode._('TT', '1868', DialCodeResolverKey.tt$);
+  static const tn = DialCode._('TN', '216', DialCodeResolverKey.tn$);
+  static const tr = DialCode._('TR', '90', DialCodeResolverKey.tr$);
+  static const tm = DialCode._('TM', '993', DialCodeResolverKey.tm$);
+  static const tc = DialCode._('TC', '1649', DialCodeResolverKey.tc$);
+  static const tv = DialCode._('TV', '688', DialCodeResolverKey.tv$);
+  static const ug = DialCode._('UG', '256', DialCodeResolverKey.ug$);
+  static const ua = DialCode._('UA', '380', DialCodeResolverKey.ua$);
+  static const ae = DialCode._('AE', '971', DialCodeResolverKey.ae$);
+  static const gb = DialCode._('GB', '44', DialCodeResolverKey.gb$);
+  static const um = DialCode._('UM', '246', DialCodeResolverKey.um$);
+  static const uy = DialCode._('UY', '598', DialCodeResolverKey.uy$);
+  static const uz = DialCode._('UZ', '998', DialCodeResolverKey.uz$);
+  static const vu = DialCode._('VU', '678', DialCodeResolverKey.vu$);
+  static const ve = DialCode._('VE', '58', DialCodeResolverKey.ve$);
+  static const vn = DialCode._('VN', '84', DialCodeResolverKey.vn$);
+  static const vg = DialCode._('VG', '1284', DialCodeResolverKey.vg$);
+  static const vi = DialCode._('VI', '1340', DialCodeResolverKey.vi$);
+  static const wf = DialCode._('WF', '681', DialCodeResolverKey.wf$);
+  static const ye = DialCode._('YE', '967', DialCodeResolverKey.ye$);
+  static const zm = DialCode._('ZM', '260', DialCodeResolverKey.zm$);
+  static const zw = DialCode._('ZW', '263', DialCodeResolverKey.zw$);
+}
+
+/// List of country information.
+///
+/// Countries are listed in alphabetical order according to the default
+/// translations, not the country code. However, countryCodes will be ordered
+/// according to the i18n content at runtime to account for custom translations.
+const dialCodes = [
```

Comment:
I think we should make DialCode an enum. You could then remove `dialCodes` in favor of `DialCode.values`. It shouldn't change much else.

DialCode as an enum.
```dart
enum DialCode {
  us$('US', '1', DialCodeResolverKey.us$),
  af$('AF', '93', DialCodeResolverKey.af$);
  // remaining codes

  const DialCode(this.dialCode, this.value, this.key);

  final String dialCode;
  final String value;
  final DialCodeResolverKey key;
}
```

Example use of `DialCodes`.values instead of `dialCodes`
```dart
myFunction() {
  print(DialCodes.values.length);
}